app/views/dishes/show.html.haml

```diff
@@ -14,4 +14,5 @@ Menu(s):
 - @dish.menus.each do |menu|
   = menu.title
 %br/
-= link_to 'Add to cart', dish_add_item_path(@dish), method: :post, class: "btn btn-success"
+- unless !current_user.nil? && current_user.role == 'owner'
```

Comment:
I don't agree ;-) Pushed up a commit that uses ability (we can revert if that is not a solution you approve)